bluemix/src/main/java/com/ibm/watson/SpeechToText.java

```diff
@@ -43,120 +39,157 @@ public String speechToText(byte[] incomingBinaryPOSTData) {
-		.getJsonObject(0).getJsonObject("credentials");
 
-            try {
-                serviceURL = credentials.getString("url");
-            } catch (Exception e) {}
-            // If we didn't find a URL for the service in VCAP_SERVICES,
-            // use the original.
+
+            serviceURL = credentials.getString("url");
+
             if ("".equals(serviceURL)) {
                 serviceURL = ORIGINAL_SERVICE_URL;
             }
-            
-	    String serviceUser = credentials.getString("username");
-	    String servicePass = credentials.getString("password");
-	    String auth = serviceUser + ":" + servicePass;
-
-	    // Prepare the query string parameters.
-	    // See the Speech to Text service documentation for details.
-	    String queryString = "";
-	    queryString = queryString + "&model=en-US_NarrowbandModel";
-	    queryString = queryString + "&continuous=true";
-
-	    if (DEBUG) {
-		System.out.println("POSTing [" + incomingBinaryPOSTData.length + "] bytes");
-		System.out.println("Using URL [" + serviceURL + API_PATH + "?" + queryString + "]");
-	    }
-
-	    // Prepare the HTTP connection to the service
-	    conn = (HttpURLConnection) new URL(serviceURL + API_PATH + "?" + queryString).openConnection();
-            conn.setDoInput(true);
-            conn.setDoOutput(true);
-            conn.setRequestMethod("POST");
-	    conn.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String(auth.getBytes()));
-            conn.setRequestProperty("Accept", "*/*");
-            conn.setRequestProperty("Content-Type", "audio/flac");
-            conn.setRequestProperty("Transfer-encoding", "chunked");
-	    conn.setFixedLengthStreamingMode(incomingBinaryPOSTData.length);
-
-	    // Uncomment the following to opt out of improving the service with this audio file
-	    //	    conn.setRequestProperty("X-WDC-PL-OPT-OUT", "1");
-            
+
+
+            // set up our http request
+            conn = setupHttpConnection(incomingBinaryPOSTData, serviceURL, credentials);
+
             // make the connection
             conn.connect();
-            
-            // send the POST request
-            OutputStream output = conn.getOutputStream();
-            output.write(incomingBinaryPOSTData);
-            output.flush();
-	    output.close();
-            
+
+            sendRequest(incomingBinaryPOSTData, conn);
+
             // Read the response from the service
-            BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), "UTF-8"));
+            String lines = readResponse(conn);
 
-	    if (DEBUG) {
-		System.out.println("Service response code: " + conn.getResponseCode());
-		System.out.println("Service response message: " + conn.getResponseMessage());
-	    }
+            // Parse the transcript string out of the Speech To Text service's JSON response
+            String transcript = readTranscripts(readJsonObject(new StringReader(lines)));
```

Comment:
Would it be more clear to move "readJsonObject(new StringReader(lines))" out of the function call and into the function?